### PR TITLE
Changelog for 3.12.0 beta

### PR DESCRIPTION
**Pull request by Lex-2008**
and 3.11.0 which should be on master

### CODE REVIEW

**Comment by olehermanse** [comment]
We should mention in HACKING.md that changelog entries in commits should be past tense. I.e. Added, Changed, Deprecated, Removed, Fixed. And maybe link to http://keepachangelog.com/en/1.0.0/ as a guideline. (Thanks @nickanderson for link)

File: CHANGELOG.md [hunk 1]
```diff
@@ -1,6 +1,132 @@
...
+	- schedule in body executor control can now be configured via augments
+	  (CFE-2508)
+	- Include scheduled report assets in self maintenance (ENT-3558)
+	- Remove unused body action aggregator and body file_select folder
+	- Remove unused body process_count check_process
+	- prevent yum from locking in package_methods when possible
+	  (CFE-2759)
+	- Render variables tagged for inventory from agent host_info_report
+	  (CFE-2750)
+	- make apt_get package module work with repositories containing spaces in the label
```
Comment (olehermanse) [inline CHANGELOG.md]: Capitalize first letter.

File: CHANGELOG.md [hunk 2]
```diff
@@ -1,6 +1,132 @@
...
+	- Policy Release Id is now inventoried by default (CFE-2097)
+	- Fix to frequent logging of enterprise license utilization (ENT-3390)
+	- Maintain access to exported CSV reports in older versions (ENT-3572)
+	- cf-execd service override template now only kills cf-execd on stop
+	  (ENT-3395)
+	- Fix self upgrade for hosts older than 3.7.4 (ENT-3368)
+	- Avoid self upgrade from triggering during bootstrap (ENT-3394)
+	- Add json templates for rendering serial and multiline data (CFE-2713)
+	- Removed unused libraries and controls
+	- fixed an error in the file_make_mustache_*, incorrect variable name used
```
Comment (olehermanse) [inline CHANGELOG.md]: Capitalize first letter.

File: CHANGELOG.md [hunk 3]
```diff
@@ -1,6 +1,132 @@
 # Changelog
 Notable changes to the framework should be documented here
 
+3.12.0b1:
+	- Avoid executing self upgrade policy unnecessarily (ENT-3592)
+	- Add amazon_linux class to yum package module
+	- Introduce ability to set policy update bundle via augments (CFE-2687)
+	- Localize delete tidy in ha update policy (ENT-3659)
```
Comment (olehermanse) [inline CHANGELOG.md]: I don't understand what this means.

File: CHANGELOG.md [hunk 4]
```diff
@@ -1,6 +1,132 @@
...
+	- Removed unused libraries and controls
+	- fixed an error in the file_make_mustache_*, incorrect variable name used
+	  (CFE-2714)
+
+3.11.0:
+	- Rename enable_client_initiated_reporting to client_initiated_reporting_enabled
+	- Directories for ubuntu 16 and centos 7 should exist in master_software_updates
+	  (ENT-3136)
+	- Fix: Automatic client upgrades for deb hosts
+	- Add aix OOTB oslevel inventory (ENT-3117)
```
Comment (olehermanse) [inline CHANGELOG.md]: Capital AIX.

File: CHANGELOG.md [hunk 5]
```diff
@@ -1,6 +1,132 @@
...
+	- Support enablerepo and disablerepo options in yum package_module
+	  (CFE-2806)
+	- Fix cf-runagent during 3.7.x -> 3.10.x migration
+	  (CFE-2776, CFE-2781, CFE-2782)
+	- Makes it possible to tune policy master_location via augments in update policy
+	  (ENT-3692)
+	- Fix inventory for total memory on AIX (CFE-2797)
+	- Do not manage redis since it's no longer used (ENT-2797)
+
+3.12.0a1:
```
Comment (nickanderson) [inline CHANGELOG.md]: We didnt release alpha1, so I think this should just be listed in the b01 section.

**Comment by nickanderson** [comment]
Looks ok, but I think we should add an entry about pg-upgrade for package upgrades.

File: CHANGELOG.md (hunk 3 above)
Comment (Lex-2008) [reply CHANGELOG.md]: @nickanderson: ? Should I remove it or can you reword it in more understandable way? I believe "ha" means "HA", which is short for "High Availability"

**Comment by Lex-2008** [comment]
I believe entry about using pg-upgrade for package upgrades should go only into Enterprise changelog, not into core or masterfiles.

File: CHANGELOG.md (hunk 3 above)
Comment (olehermanse) [reply CHANGELOG.md]: Yes, I know what HA means, and yes it should be uppercase.

> Localize delete tidy

Still doesn't make much sense to me.

**Pull request merged**